### DISCUSSION TITLE
Installation ERROR: Process exited with code 25

### DISCUSSION THREAD

**Discussion by mitenheoh** [MiSchroe/ioBroker.klf200#115] [Q&A] [answered]
Hallo Experten,
die Installation von GitHub scheitert mit dem unten angehängten Log.
Meine Installation von ioBroker (Synology NAS, Protainer, Docker) ist frisch aktualisiert mit Container-Status "healthy": 
Node.js        18.17.1
NPM            9.6.7
js-controller 5.0.12
admin          6.10.1

Wer kann mir Hinweise geben, wo ich ansetzen kann? Oder ist der Adapter klf200 derzeit veraltet? Ist er tot?

Vielen Dank
Tim

$ iobroker url MiSchroe/ioBroker.klf200 --host iobroker2023 --debug

install MiSchroe/ioBroker.klf200

NPM version: 9.6.7

Installing MiSchroe/ioBroker.klf200... (System call)

npm ERR! code 1

npm ERR! npm ERR! npm ERR! > iobroker.klf200@1.0.1 buildnpm ERR! > build-adapter typescriptnpm ERR! npm ERR! npm ERR! Compiling TypeScript with ESBuild...npm ERR! npm ERR! Type-checking ./tsconfig.build.json with tsc...npm ERR! error TS2688: Cannot find type definition file for 'bl'.npm ERR! The file is in the program because:npm ERR! Entry point for implicit type library 'bl'npm ERR! npm WARN using --force Recommended protections disabled.npm ERR! npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecatednpm ERR! npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecatednpm ERR! npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecatednpm ERR! npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecatednpm ERR! npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependenciesnpm ERR! npm WARN deprecated uuid@3.4.0: Please upgrade to version 7 or higher. Older versions may use Math.random() in certain circumstances, which is known to be problematic. See https://v8.dev/blog/math-random for details.npm ERR! npm WARN using --force Recommended protections disabled.npm ERR! npm ERR! build/setupProducts.js 20.8kbnpm ERR! build/main.js 11.1kbnpm ERR! build/setup.js 9.3kbnpm ERR! build/setupGroups.js 9.1kbnpm ERR! build/util/propertyLink.js 7.8kbnpm ERR! build/setupScenes.js 7.0kbnpm ERR! build/util/converter.js 4.3kbnpm ERR! build/lib/tools.js 2.3kbnpm ERR! build/util/utils.js 1.5kbnpm ERR! build/util/promiseQueue.js 1.3kbnpm ERR! build/util/stateHelper.js 1.3kbnpm ERR! build/setupProducts.js.map 31.8kbnpm ERR! build/main.js.map 21.2kbnpm ERR! build/setup.js.map 15.2kbnpm ERR! build/setupGroups.js.map 13.6kbnpm ERR! build/util/propertyLink.js.map 12.1kbnpm ERR! build/setupScenes.js.map 10.8kbnpm ERR! build/util/converter.js.map 4.2kbnpm ERR! build/lib/tools.js.map 2.1kbnpm ERR! build/util/utils.js.map 917b npm ERR! ...and 2 more output files...npm ERR! npm ERR! ⚡ Done in 37msnpm ERR! ❌ Type-checking ./tsconfig.build.json failed!npm ERR! npm ERR! code 1npm ERR! npm ERR! path /opt/iobroker/.npm/_cacache/tmp/git-cloneTki1Denpm ERR! npm ERR! command failednpm ERR! npm ERR! command sh -c npm run buildnpm ERR! npm ERR! npm ERR! A complete log of this run can be found in: /opt/iobroker/.npm/_logs/2023-09-27T12_40_26_407Z-debug-0.log

npm ERR! A complete log of this run can be found in: /opt/iobroker/.npm/_logs/2023-09-27T12_40_14_448Z-debug-0.log

npm ERR! code 1npm ERR! git dep preparation failednpm ERR! command /usr/bin/node /usr/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/opt/iobroker/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit --include=dev --include=peer --include=optional --no-package-lock-only --no-dry-runnpm ERR! > iobroker.klf200@1.0.1 preparenpm ERR! > npm run buildnpm ERR! npm ERR! npm ERR! > iobroker.klf200@1.0.1 prebuildnpm ERR! > rimraf ./buildnpm ERR! npm ERR! npm ERR! > iobroker.klf200@1.0.1 buildnpm ERR! > build-adapter typescriptnpm ERR! npm ERR! npm ERR! Compiling TypeScript with ESBuild...npm ERR! npm ERR! Type-checking ./tsconfig.build.json with tsc...npm ERR! error TS2688: Cannot find type definition file for 'bl'.npm ERR! The file is in the program because:npm ERR! Entry point for implicit type library 'bl'npm ERR! npm WARN using --force Recommended protections disabled.npm ERR! npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecatednpm ERR! npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecatednpm ERR! npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecatednpm ERR! npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecatednpm ERR! npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependenciesnpm ERR! npm WARN deprecated uuid@3.4.0: Please upgrade to version 7 or higher. Older versions may use Math.random() in certain circumstances, which is known to be problematic. See https://v8.dev/blog/math-random for details.npm ERR! npm WARN using --force Recommended protections disabled.npm ERR! npm ERR! build/setupProducts.js 20.8kbnpm ERR! build/main.js 11.1kbnpm ERR! build/setup.js 9.3kbnpm ERR! build/setupGroups.js 9.1kbnpm ERR! build/util/propertyLink.js 7.8kbnpm ERR! build/setupScenes.js 7.0kbnpm ERR! build/util/converter.js 4.3kbnpm ERR! build/lib/tools.js 2.3kbnpm ERR! build/util/utils.js 1.5kbnpm ERR! build/util/promiseQueue.js 1.3kbnpm ERR! build/util/stateHelper.js 1.3kbnpm ERR! build/setupProducts.js.map 31.8kbnpm ERR! build/main.js.map 21.2kbnpm ERR! build/setup.js.map 15.2kbnpm ERR! build/setupGroups.js.map 13.6kbnpm ERR! build/util/propertyLink.js.map 12.1kbnpm ERR! build/setupScenes.js.map 10.8kbnpm ERR! build/util/converter.js.map 4.2kbnpm ERR! build/lib/tools.js.map 2.1kbnpm ERR! build/util/utils.js.map 917b npm ERR! ...and 2 more output files...npm ERR! npm ERR! ⚡ Done in 37msnpm ERR! ❌ Type-checking ./tsconfig.build.json failed!npm ERR! npm ERR! code 1npm ERR! npm ERR! path /opt/iobroker/.npm/_cacache/tmp/git-cloneTki1Denpm ERR! npm ERR! command failednpm ERR! npm ERR! command sh -c npm run buildnpm ERR! npm ERR! npm ERR! A complete log of this run can be found in: /opt/iobroker/.npm/_logs/2023-09-27T12_40_26_407Z-debug-0.lognpm ERR! A complete log of this run can be found in: /opt/iobroker/.npm/_logs/2023-09-27T12_40_14_448Z-debug-0.log

host.iobroker2023 Cannot install MiSchroe/ioBroker.klf200: 1

ERROR: Process exited with code 25

**Comment by MiSchroe** [comment, marked as answer]
Hallo Tim,

installiere den Adapter entweder über die Web-Oberfläche oder über `iobroker add klf200`.

**Reply by mitenheoh** [reply]
Vielen Dank Michael,
jetzt hat es auf Anhieb geklappt - und ich konnte es ganz normal unter den verfügbaren Adaptern finden, musste also garnicht im Expertenmodus von GitHub installieren.
